### PR TITLE
Feat : Added addToWatchLater and removeFromWatchlater Functionality

### PR DESCRIPTION
**Pull request by shlok30**
Hi Team,

I have added addToWatchlater and removeFromWatchlater functionality to my app for now.

Auth has not been setup completely yet, so please use the following dummy credentials to try out the functionality

email: test@gmail.com
password: test123

Live Link : https://water-videos-f6is3zje2-shlok30.vercel.app/

Demo Video

https://user-images.githubusercontent.com/60795588/168979494-f71fa507-8a10-48c4-81df-e44b006da4d7.mov

Please review my code and let me know if any changes are required?

Thanks in advance

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {createContext , useContext, useState} from 'react'
+import authFunction from './auth-functions/authFunction'
+
+const AuthContext = createContext()
+
+const AuthContextProvider = ({children}) => {
+
+    const [email , setEmail] = useState("")
+    const [password, setPassword] = useState("")
```
Comment (ivinayakg) [inline src/context/auth-context.js:9]: No need to store password in the context.

File: src/pages/watch-later.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    return(
+        <>
+            <div className="flex">
+                <div className="grow-1">
+                    <Sidebar />
+                </div>
+                <div className="container center-block">
+                    <h2 className = "h-l m5-top">Watch Later</h2>
+                    <div className="flex gap-m m3-top self-start">
+                        {renderVideoCards(watchlater)}
```
Comment (ivinayakg) [inline src/pages/watch-later.js:18]: Nice.

**Review by ivinayakg — APPROVED** [review]
I really like that you have made the code clean and simple, great work 👍

File: src/context/reducers/user-reducer.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+const userReducer = (state,{type,payload}) => {
+    switch(type){
+        case "LOGIN":
+            console.log("Login was used")
```
Comment (sharmayatendra) [inline src/context/reducers/user-reducer.js:5]: remove console

File: src/index.js [hunk 4]
```diff
@@ -5,6 +5,8 @@ import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
 import { VideoContextProvider } from "./context/video-context";
+import { AuthContextProvider } from "./context/auth-context";
+import { UserContextProvider } from "./context/user-context";
```
Comment (sharmayatendra) [inline src/index.js:9]: you can import all context from a single file

File: src/pages/signup.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                <div className = "container auth-container center-block" style={{border:"1px solid black"}}>
+                    <div className = "primary-text center-text text-m bold primary-colour">Signup</div>
+                    <div className = "padding-s">
+                        <label htmlFor="email-field" className="input-label">Email Address</label>
+                        <input type = "email" id = "email-field" className = "input-field m2-top full-width" style = {{display: "block",width:"100%"}} placeholder="Please Enter Email Here" value = {email} onChange = {(e) => setEmail(e.target.value)}/>
+                    </div>
+
+                    <div className="padding-s flex flex-column">
+                        <label htmlFor="password-field" className="input-label">Enter Password</label>
+                        <input type = "password" id = "password-field" className = "input-field m2-top full-width" style = {{display: "block",width:"100%"}} placeholder="Please Enter Password Here" value = {password} onChange={(e) => setPassword(e.target.value)} />
```
Comment (sharmayatendra) [inline src/pages/signup.js:25]: try to avoid inline styles

**Review by sharmayatendra — APPROVED** [review]
good work .. clean code.
you can look upon suggested changes

**Pull request merged**